js/views/message.js

```diff
@@ -277,30 +277,16 @@ views.Messages = Backbone.Marionette.CompositeView.extend({
 			.val(t('mail', 'Loading …'))
 			.prop('disabled', true);
 
-		var url = OC.generateUrl(
-			'apps/mail/accounts/{accountId}/folders/{folderId}/messages?from={from}&to={to}',
-			{
-				'accountId': Mail.State.currentAccountId,
-				'folderId':Mail.State.currentFolderId,
-				'from': from,
-				'to': from + 20
-			});
-		if (this.filterCriteria) {
-			url = OC.generateUrl(
-				'apps/mail/accounts/{accountId}/folders/{folderId}/messages?filter={query}&from={from}&to={to}',
-				{
-					'accountId': Mail.State.currentAccountId,
-					'folderId':Mail.State.currentFolderId,
-					'query': this.filterCriteria.text,
-					'from': from,
-					'to': from + 20
-				});
-		}
 		var self = this;
-		$.ajax(url, {
-				data: {},
-				type:'GET',
-				success: function (jsondata) {
+		Mail.Communication.fetchMessageList(
```

Comment:
with respect to modularity and testability (one day - LOL) I guess we should inject the Communications object into the view ..... one day ...